tests/poll.rs

```diff
@@ -201,6 +208,94 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
+#[test]
+fn registry_operations_are_thread_safe() {
+    init();
+
+    let (mut poll, mut events) = init_with_poll();
+
+    let registry = Arc::new(poll.registry().try_clone().unwrap());
+    let registry1 = Arc::clone(&registry);
+    let registry2 = Arc::clone(&registry);
+    let registry3 = Arc::clone(&registry);
+
+    let barrier = Arc::new(Barrier::new(4));
+    let barrier1 = Arc::clone(&barrier);
+    let barrier2 = Arc::clone(&barrier);
+    let barrier3 = Arc::clone(&barrier);
+
+    let mut listener = TcpListener::bind(any_local_address()).unwrap();
+    let addr = listener.local_addr().unwrap();
+
+    let mut udp_socket = UdpSocket::bind(any_local_address()).unwrap();
+
+    /* Expect that multiple register/deregister/reregister work fine on multiple
+    threads. Main thread will wait before the expect_events for all other 3
+    threads to do their work. Otherwise expect_events timeout might be too short
+    for all threads to complete, and call might fail. */
+
+    let handle1 = thread::spawn(move || {
+        registry_ops_flow(
+            &registry1,
+            &mut listener,
+            ID1,
+            Interest::READABLE,
+            Interest::READABLE,
+        )
+        .unwrap();
+
+        barrier1.wait();
+        barrier1.wait();
+    });
+
+    let handle2 = thread::spawn(move || {
+        registry_ops_flow(
+            &registry2,
+            &mut udp_socket,
+            ID2,
+            Interest::READABLE,
+            Interest::READABLE,
+        )
+        .unwrap();
+
+        barrier2.wait();
+        barrier2.wait();
+    });
+
+    let handle3 = thread::spawn(move || {
+        let mut stream = TcpStream::connect(addr).unwrap();
+        registry_ops_flow(
+            &registry3,
+            &mut stream,
+            ID3,
+            Interest::READABLE,
+            Interest::READABLE | Interest::WRITABLE,
+        )
+        .unwrap();
+
+        barrier3.wait();
+        barrier3.wait();
+    });
+
+    // wait for threads to finish before expect_events
+    barrier.wait();
+    expect_events(
+        &mut poll,
+        &mut events,
+        vec![
+            ExpectEvent::new(ID1, Interest::READABLE),
```

Comment:
No writable for the `UdpSocket`?

Reply:
done, added writable instead of readable

Reply:
I still don't see an `ExpectEvent` for `ID2`.

Reply:
done, added, thanks